src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/OptionSetInfo.cs

```diff
@@ -2,13 +2,18 @@
 // Licensed under the MIT license.
 
 using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
 using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.IR;
 using Microsoft.PowerFx.Core.Localization;
 using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Functions;
+using Microsoft.PowerFx.Types;
 
-namespace Microsoft.PowerFx.Core.Texl.Builtins
+namespace Microsoft.PowerFx.Functions
 {
-    internal sealed class OptionSetInfoFunction : BuiltinFunction
+    internal sealed class OptionSetInfoFunction : BuiltinFunction, IAsyncTexlFunction
```

Comment:
We should not have any function implementation inside Microsoft.PowerFx.Core. This should be moved to the MS.PFx.Interpreter instead

Reply:
Done.